File: student.cpp
#ifndef STUDENT_H
#define STUDENT_H

#include <iostream>
#include <vector>
using namespace std; // la bibliotheque de goat 

class Student {
private:
    string name; // Nom de l'étudiant
    int studentID; // Numéro d’identification unique
    int age; // Âge de l’étudiant
    vector<double> grades; // Liste des notes obtenues

public:
    // Constructeur
    Student(string name, int id, int age);

    // Méthodes
    void addGrade(double grade); // Ajoute une note à l'étudiant
    double getAverageGrade(); // Retourne la moyenne des notes
    void display(); // Affiche les infos de l'étudiant
    int getID(); // Retourne l'ID de l'étudiant
};

#endif

File: student.py
## student.py
from person import Person # import de la class person 

class Student(Person): # elle herite de person donc elle posede c attribu 
    def __init__(self, name, studentID, age): # dabord person 
        super().__init__(name, age) #
        self._studentID = studentID
        self._grades = []
    
    def add_grade(self, grade): # on ajoute une note a la liste grade 
        self._grades.append(grade)
    
    def get_average_grade(self): # 
        return sum(self._grades) / len(self._grades) if self._grades else 0

## undergraduate_student.py
from student import Student

class UndergraduateStudent(Student):
    pass

## graduate_student.py
from student import Student

class GraduateStudent(Student):
    def get_average_grade(self):
        return (super().get_average_grade() * 1.1)  # Example of polymorphism